lib/mastodon/cli/main.rb

```diff
@@ -92,55 +91,37 @@ def self_destruct
-
-      unless dry_run?
-        prompt.warn('This operation WILL NOT be reversible. It can also take a long time.')
-        prompt.warn('While the data won\'t be erased locally, the server will be in a BROKEN STATE afterwards.')
-        prompt.warn('A running Sidekiq process is required. Do not shut it down until queues clear.')
-
-        exit(1) if prompt.no?('Are you sure you want to proceed?')
-      end
+      if SelfDestructHelper.self_destruct?
+        prompt.ok('Self-destruct mode is already enabled for this Mastodon server')
 
-      inboxes   = Account.inboxes
-      processed = 0
+        pending_accounts = Account.local.without_suspended.count + Account.local.suspended.joins(:deletion_request).count
+        sidekiq_stats = Sidekiq::Stats.new
 
-      Setting.registrations_mode = 'none' unless dry_run?
+        if pending_accounts.positive?
+          prompt.warn("#{pending_accounts} accounts are still pending deletion.")
+        elsif sidekiq_stats.enqueued.positive?
+          prompt.warn('Deletion notices are still being processed')
+        elsif sidekiq_stats.retry_size.positive?
+          prompt.warn('At least one delivery attempt for each deletion notice has been made, but some have failed and are scheduled for retry')
+        else
+          prompt.ok('Every deletion notice has been sent! You can safely delete all data and decomission your servers!')
+        end
 
-      if inboxes.empty?
-        Account.local.without_suspended.in_batches.update_all(suspended_at: Time.now.utc, suspension_origin: :local) unless dry_run?
-        prompt.ok('It seems like your server has not federated with anything')
-        prompt.ok('You can shut it down and delete it any time')
-        return
+        exit(0)
       end
 
-      prompt.warn('Do NOT interrupt this process...')
-
-      delete_account = lambda do |account|
-        payload = ActiveModelSerializers::SerializableResource.new(
-          account,
-          serializer: ActivityPub::DeleteActorSerializer,
-          adapter: ActivityPub::Adapter
-        ).as_json
-
-        json = Oj.dump(ActivityPub::LinkedDataSignature.new(payload).sign!(account))
-
-        unless dry_run?
-          ActivityPub::DeliveryWorker.push_bulk(inboxes, limit: 1_000) do |inbox_url|
-            [json, account.id, inbox_url]
-          end
-
-          account.suspend!(block_email: false)
-        end
+      exit(1) unless prompt.ask('Type in the domain of the server to confirm:', required: true) == Rails.configuration.x.local_domain
 
-        processed += 1
-      end
+      prompt.warn('This operation WILL NOT be reversible. It can also take a long time.')
+      prompt.warn('While the data won\'t be erased locally, the server will be in a BROKEN STATE afterwards.')
+      prompt.warn('A running Sidekiq process is required. Do not shut it down until queues clear.')
```

Comment:
I am not sure this is worthwhile, as this is information that may be difficult to keep up to date. In contrast, re-running the command will give the admin the state of the self-destruct process. This is stated at the end of the command, but maybe we could state it earlier.

Reply:
No issue with this, but maybe reword "a sidekiq process". Because if you run multiple sidekiq processes, each processing different queues, then you need to keep them all active.